samples/tiny_http/src/main.rs

```diff
@@ -1,43 +1,56 @@
 extern crate tiny_http;
 extern crate multipart;
 
+use std::fs::File;
+use std::io::{Error, Read};
 use multipart::server::{Multipart, Entries, SaveResult};
 use tiny_http::{Response, StatusCode, Request};
 fn main() {
     // Starting a server on `localhost:80`
-    let server = tiny_http::Server::http("localhost:80").unwrap();
+    let server = tiny_http::Server::http("localhost:80").expect("Could not bind localhost:80");
     loop {
         // This blocks until the next request is received
         let mut request = server.recv().unwrap();
-        process_multipart(&mut request).unwrap();
-
-        // Answering with an HTTP OK and a string
-        let response_string = "Multipart data is received!".as_bytes();
-        let response = Response::new(StatusCode(200),
-                                     vec![],
-                                     response_string,
-                                     Some(response_string.len()),
-                                     None);
-        request.respond(response).unwrap();
+
+        // Processes a request and returns response or an occured error
+        let result = process_request(&mut request);
+        let resp = match result {
+            Ok(resp) => resp,
+            Err(e) => {
+                println!("An error has occured during request proccessing: {:?}", e);
+                build_response(500, "The received data was not correctly proccessed on the server")
+            }
+        };
+
+        // Answers with a response to a client
+        request.respond(resp).unwrap();
     }
 }
 
-fn process_multipart(request: &mut Request) -> Result<(), Error> {
+/// Processes a request and returns response or an occured error.
+fn process_request<'a, 'b>(request: &'a mut Request) -> Result<Response<&'b [u8]>, Error> {
```

Comment:
I was trying to form error messages, but I couldn't, because of lifetimes. That's why they weren't `static`.

And didn't know about the first case of elision, thanks!